cmd/relayproxy/config/config.go

```diff
@@ -91,10 +92,15 @@ func New(flagSet *pflag.FlagSet, log *zap.Logger, version string) (*Config, erro
 			log.Error("error loading file", zap.Error(errBindFile))
 		}
 	}
-
+	configMap := k.Raw()
 	// Map environment variables
-	_ = k.Load(env.Provider("", ".", func(s string) string {
-		return strings.ReplaceAll(strings.ToLower(s), "_", ".")
+	_ = k.Load(env.ProviderWithValue("", ".", func(s string, v string) (string, interface{}) {
+		if strings.HasPrefix(s, "RETRIEVERS") || strings.HasPrefix(s, "NOTIFIERS") {
```

Comment:
```suggestion
		if strings.HasPrefix(s, "RETRIEVERS") || strings.HasPrefix(s, "NOTIFIERS") {
			configMap := k.Raw()
```

We move the `configMap` variable inside the function.

Reply:
Hi @thomaspoignant, I don't think so, I have reloaded it every time. I also have ut to cover it
https://github.com/thomaspoignant/go-feature-flag/pull/1902/files#diff-9b365e54858c1fbb3914851507d0d67c4b848f841db79c0f460d18f2df974926R387

Reply:
Your test does not cover what I wanted to highlight.
If in you test you have 2 entries for the same number (as proposed [here](https://github.com/thomaspoignant/go-feature-flag/pull/1902/files#r1616765918)), you will see that only the last env variable will be taken in consideration.

For example here, you will have only the `RETRIEVERS_2_HEADERS_X-GOFF-CUSTOM` in your configuration and not `RETRIEVERS_2_HEADERS_AUTHORIZATION`.

If you move the `configMap` variable as suggested it will work.